solar-system.rb

```diff
@@ -0,0 +1,200 @@
+class SolarSystem
+  attr_accessor :system
+  # Get array of Planets
+  def initialize(planets)
+    @solar_system = planets
+  end
+
+  # Show an initial menu of Planets from which to choose
+  def show_menu
+    planet_list = ""
+    i = 1
+    @solar_system.length.times do |n|
+      planet_list << "\n#{i}. #{@solar_system[n].name}"
+      i += 1
+    end
+    return "\nPlanet List#{planet_list}\nOther: Exit\n\n"
+  end
+
+  # Show the selected planet's information
+  def show_planet(number)
+    selected = @solar_system[number-1]
+
+    # Calculate planet related ages
+    age = 4.5e9/(selected.rotation_time/12.0)
+    age = age.to_i.to_s
+    age = age.reverse.scan(/\d{3}|.+/).join(",").reverse
+
+    # Calculate the closest neighboring planets
+    # Collect names and distances of planets
+    distance_and_name = []
+    @solar_system.length.times do |i|
+      planet_info = {}
+      planet_info[:name] = @solar_system[i].name
```

Comment:
You could use an `each` loop here! These three lines would then look like:

```ruby'
@solar_system.each do |planet|
  planet_info = {}
  planet_info[:name] = planet.name
```

This might be a little more readable.